src/Servers/Kestrel/Core/src/Internal/AddressBinder.cs

```diff
@@ -142,18 +142,17 @@ internal static ListenOptions ParseAddress(string address, out bool https)
 
         return options;
 
-        bool IsLocalhost(string host, out string? prefix)
+        static bool IsLocalhost(string host, out string? prefix)
         {
             // Check for "localhost"
-            if (string.Equals(parsedAddress.Host, "localhost", StringComparison.OrdinalIgnoreCase))
+            if (string.Equals(host, "localhost", StringComparison.OrdinalIgnoreCase))
             {
                 prefix = null;
                 return true;
             }
 
             // Check for use of .localhost TLD (Top Level Domain)
-            if (host.EndsWith(".localhost", StringComparison.OrdinalIgnoreCase)
-                && !string.Equals(parsedAddress.Host, ".localhost", StringComparison.OrdinalIgnoreCase))
+            if (host.Length > 10 && host.EndsWith(".localhost", StringComparison.OrdinalIgnoreCase))
```

Comment:
Maybe 10 should become a const.

Reply:
If you wanted to do that, then you could do this:

```csharp
const string Suffix = ".localhost";
if (host.Length > Suffix.Length && host.EndsWith(Suffix, StringComparison.OrdinalIgnoreCase))
{
    // Code
}
```